### DISCUSSION TITLE
SQLServer TPCC TpmC calculation errors

### DISCUSSION THREAD

**Discussion by mkprabhakaran** [TPC-Council/HammerDB#288] [General]
When TPCC workload on SQLServer is overloaded for example 2000 vUsers are run against a small database, HammerDB doesnt stop counting at the end of the test period. Instead it waits for the entire workload to complete, and counts all the transactions that are completed. It doesnt take into account query response time.

**Comment by sm-shaw** [comment]
Yes, HammerDB does not arbitrarily limit the number of virtual users to warehouses, allowing the user the discretion to use best practice.  Guidelines on virtual users to warehouses are here https://www.hammerdb.com/blog/uncategorized/how-many-warehouses-for-the-hammerdb-tpc-c-test/ and here https://www.hammerdb.com/docs/ch03s07.html and of course the number of warehouses and virtual users to configure is based on the system under test for which you don't provide any information. If the CPU is saturated, then of course HammerDB will not be able to run the queries to capture the NOPM and TPM values and will take a longer time to receive the notification that the workload has stopped. This is expected behaviour, it could be prevented by imposing an arbitrary VU limit to warehouse, however understanding the workload and allowing a flexible configuration is a better approach for the majority of users.
